### PR TITLE
Improve KORE parser performance

### PR DESCRIPTION
**Pull request by tothtamas28**
* Remove `_LookaheadBuffer`
* Remove `KoreStringLexer`
* Use `dequote_str` in parser
* Other small simplifications

### CODE REVIEW

**Review by ehildenb — APPROVED** [review]
I'm approving, but I don't feel very comfortable reviewing parser code. It may be best to get @radumereuta to also have a look, because I only skimmed.

File: src/pyk/kore/parser.py [hunk 1]
```diff
@@ -75,373 +74,342 @@
...
-            KoreToken.Type.KW_AXIOM: self.axiom,
-            KoreToken.Type.KW_CLAIM: self.claim,
+            TokenType.KW_IMPORT: self.importt,
+            TokenType.KW_SORT: self.sort_decl,
+            TokenType.KW_HOOKED_SORT: self.hooked_sort_decl,
+            TokenType.KW_SYMBOL: self.symbol_decl,
+            TokenType.KW_HOOKED_SYMBOL: self.hooked_symbol_decl,
+            TokenType.KW_ALIAS: self.alias_decl,
+            TokenType.KW_AXIOM: self.axiom,
+            TokenType.KW_CLAIM: self.claim,
```
Comment (ehildenb) [inline src/pyk/kore/parser.py:119]: Same here?

File: src/pyk/kore/parser.py [hunk 2]
```diff
@@ -75,373 +74,342 @@
...
+            TokenType.ML_NU: self.nu,
+            TokenType.ML_CEIL: self.ceil,
+            TokenType.ML_FLOOR: self.floor,
+            TokenType.ML_EQUALS: self.equals,
+            TokenType.ML_IN: self.inn,
+            TokenType.ML_NEXT: self.next,
+            TokenType.ML_REWRITES: self.rewrites,
+            TokenType.ML_DV: self.dv,
+            TokenType.ML_LEFT_ASSOC: self.left_assoc,
+            TokenType.ML_RIGHT_ASSOC: self.right_assoc,
```
[the remark this answers is not in the document]
Comment (tothtamas28) [reply src/pyk/kore/parser.py:108]: Good point, this can be eliminated with dynamic method access.

[the remark this answers is not in the document]
Comment (tothtamas28) [reply src/pyk/kore/parser.py]: Done.

File: src/pyk/kore/parser.py (hunk 1 above)
Comment (tothtamas28) [reply src/pyk/kore/parser.py]: Done.

**Review by radumereuta — APPROVED** [review]
lgtm.
My confidence is only medium, though. But it's mostly refactoring.
I would have chosen an off the shelf parser generator instead of implementing one from scratch.
The grammar would have been easier to read, and the performance would have been already worked out.

Anyway, @tothtamas28 you say that you improved performance. Do you have any numbers? Just to get a rough idea of the magnitude: scaling with size, removing hotspots?

**Comment by tothtamas28** [comment]
> you say that you improved performance. Do you have any numbers?

On `src/tests/profiling/test-data/definition.kore` (which is a kompiled KEVM-Foundry definition), it's around 30% faster.

**Comment by radumereuta** [comment]
I was hoping for more info...
Feel free to merge whenever you are ready.

**Pull request merged**